kms: Use VK_EXT_external_memory_dmabuf to create dmabuf-backed Vulkan images

afrantzis opened the pull request:
Drop the vkCreateDmaBufImageINTEL extension function and use the functionality provided by the official VK_EXT_external_memory_dmabuf extension to create dmabuf-backed Vulkan images.

JeffyCN commented:
Hi afrantzis,

Tested on my device with ARM mali valhall GPU, it almost work, but has a tiny issue:
In the new patchset, we're using gbm_bo_get_modifier() to get bo's modifier and pass it to vk.
But the mali returned DRM_FORMAT_MOD_INVALID in my case, and rejected it in vk with a InvalidDrmFormatModifierPlaneLayoutEXTError.

Maybe we can try to add modifier info only when it is valid(not DRM_FORMAT_MOD_INVALID and having modifier extension), for example:
https://source.chromium.org/chromium/chromium/src/+/main:gpu/vulkan/vulkan_image_linux.cc;l=76

afrantzis commented:
I have merged this at https://github.com/vkmark/vkmark/commit/ab6e6f34077722d5ae33f6bd40b18ef9c0e99a15. I have incorporated a fallback (use "optimal" tiling instead of "modifier" tiling) for when we get DRM_FORMAT_MOD_INVALID, but note that this is not guaranteed to work, so I have also added a warning. Let me know how this works in your scenario.

JeffyCN commented:
it doesn't work on my board, the mali drow exception without setting the modifier_info.
setting tiling doesn't affect it, faking the modifier to linear works.

guessing it's an ARM mali issue...
